File: contract-service/src/main/java/de/beisert/examples/helvetia/contractservice/entity/Address.java
package de.beisert.examples.helvetia.contractservice.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.Data;

@Entity
@Data
public class Address implements Short{

    @Id
    @Column(name="address_id")
    private Long id;

    private String street;

    private String city;

    private String zipcode;

    public String getDisplayName() {
        return street + ", " + city;
    }

}

File: contract-ui/src/model/address.ts
export interface Address {
  id: number;
  street: string;
  city: string;
  zipcode: string;
  displayName: string;
}

export function newEmptyAddress(): Address {
  return {
    id: 0,
    street: "",
    city: "",
    zipcode: "",
    displayName: ""
  }
}
